layers/dhcpv4.go

```diff
@@ -540,21 +536,38 @@ func (o *DHCPOption) encode(b []byte) error {
 func (o *DHCPOption) decode(data []byte) error {
 	if len(data) < 1 {
 		// Pad/End have a length of 1
-		return errors.New("Not enough data to decode")
+		return DecOptionNotEnoughData
 	}
 	o.Type = DHCPOpt(data[0])
 	switch o.Type {
 	case DHCPOptPad, DHCPOptEnd:
 		o.Data = nil
 	default:
-		if len(data) < 3 {
-			return errors.New("Not enough data to decode")
+		if len(data) < 2 {
+			return DecOptionNotEnoughData
 		}
 		o.Length = data[1]
-		if o.Length > 253 {
-			return errors.New("data too long to decode")
+		if int(o.Length) > len(data[2:]) {
+			return DecOptionMalformed
 		}
-		o.Data = data[2 : 2+o.Length]
+		o.Data = data[2 : 2+int(o.Length)]
 	}
 	return nil
 }
+
+// DHCPv4Error is used for constant errors for DHCPv4. It is needed for test asserts.
+type DHCPv4Error string
+
+// DHCPv4Error implements error interface.
+func (d DHCPv4Error) Error() string {
+	return string(d)
+}
+
+const (
```

Comment:
I have two comments on this : 
1 - As the whole namespace "layers" is quite crowded, I would suggest to rename these errors with "DHCP" prefix.
2 - Adding errors as const will allocate them in any case. I know that @gconnell is looking to keep the whole footprint of gopacket low, so maybe we should not do it. @gconnell any idea on that ?